dynamodb/src/test/scala/uk/gov/nationalarchives/DADynamoDBClientTest.scala

```diff
@@ -204,4 +241,84 @@ class DADynamoDBClientTest extends AnyFlatSpec with MockitoSugar {
     }
     updateAttributeValueEx.getMessage should be("Table name could not be found")
   }
+
+  implicit val productFormat: Typeclass[MockResponse] = new Typeclass[MockResponse] {
+    override def read(av: DynamoValue): Either[DynamoReadError, MockResponse] = Right(MockSingleAttributeResponse(""))
+
+    override def write(t: MockResponse): DynamoValue = t match {
+      case MockSingleAttributeResponse(mockAttribute) =>
+        DynamoValue.fromMap(Map("mockAttribute" -> DynamoValue.fromString(mockAttribute)))
+      case MockTwoAttributesResponse(mockAttribute, mockAttribute2) =>
+        DynamoValue.fromMap(
+          Map(
+            "mockAttribute" -> DynamoValue.fromString(mockAttribute),
+            "mockAttribute2" -> DynamoValue.fromString(mockAttribute2)
+          )
+        )
+    }
+  }
+
+  val putItemsTable: TableFor2[List[MockResponse], FieldName] = Table(
+    ("input", "expectedResponse"),
+    (List(MockSingleAttributeResponse("mockValue")), "{mockAttribute=AttributeValue(S=mockValue)}"),
```

Comment:
Are the uses of `MockXXResponse`s in this `Table`, here correct? It looks different from the `getItems` one.

Reply:
Yeah, I think so. The table is different to the getItems one but it's a different request.